src/ui/public/filter_bar/editable_filter.js

```diff
@@ -0,0 +1,33 @@
+define(function (require) {
+  var _ = require('lodash');
+  var angular = require('angular');
+
+  require('ui/modules')
+  .get('kibana')
+  .directive('editableFilter', function () {
```

Comment:
Can you change this to `jsonInput` and move it out of the filter bar? This feels like a great reusable component.